### DISCUSSION TITLE
set default filter function for all columns

### DISCUSSION THREAD

**Discussion by victorkvarghese** [TanStack/table#4274] [Q&A] [answered]
Hi,

How does react-table figure out which filter function to use?

I have a table with multi select filter for all columns, filter is working with single values, but with array its failing

if i provide filter as `filterFn:'arrIncludesSome'` it works.

```
  {
    accessorKey: 'id',
    header: 'id',
    cell: info => <CEllText text={info.getValue() as string}/>,
    size: 50,
    filterFn:'arrIncludesSome'
  },

```

I think by default table uses `includesString` but i need to used `arrIncludesSome` as default filter

is there a way to override this? without defining filter function for each column def?

**UPDATE**:

I checked the filter fns code, 

seems like based on type we are choosing the filter fns. But for multi select filters `includesString` will fail

**Comment by tannerlinsley** [comment, marked as answer]
Provide a defaultColumn to the table

Tanner Linsley

**Reply by victorkvarghese** [reply]
state: {
      sorting,
      rowSelection,
      columnFilters,
    },
    defaultColumn: {
      filterFn: 'arrIncludesSome',
    },

That helped. Thanks :)

**Reply by lobaak** [reply]
This fixed my issue. Thanks!

**Comment by AskAlice** [comment]
@victorkvarghese can you provide a working example of this? I'm curious how the filter component works. At the moment It's throwing ```filterValue.some is not a function``` presumably because my filter is currently set up to filter a single text value. just not sure how to change the filter component to line up with this array type

**Reply by victorkvarghese** [reply]
@AskAlice 

You need to set filters separately. ArrIncludes some will only work with array data type. For string you need to use includesString.

Check the data types used in each column and set filter accordingly. 

My scenario was most / all of the column had arrays, while 1 or 2 were strings/booleans. So for that columns , i had to specify proper filter functions.
